ambry-rest/src/main/java/com.github.ambry.rest/NettyServerFactory.java

```diff
@@ -78,11 +80,39 @@ public NettyServerFactory(VerifiableProperties verifiableProperties, MetricRegis
   }
 
   /**
+   * Initialize the {@link ChannelHandler}s to be used in the netty pipeline
+   * @param nettyConfig the {@link NettyConfig} instance that defines the configuration parameters for the NettyServer.
+   * @param nettyMetrics the {@link NettyMetrics} instance to use to record metrics.
+   * @param requestHandler the {@link RestRequestHandler} that can be used to submit requests that need to be handled.
+   * @param publicAccessLogger the {@link PublicAccessLogger} that can be used for public access logging
+   * @param restServerState the {@link RestServerState} that can be used to check the health of the system
+   *                              to respond to health check requests
+   */
+  private void initializeChannelHandlers(NettyMetrics nettyMetrics, NettyConfig nettyConfig,
```

Comment:
The purpose of this patch is to introduce flexibility when creating pipeline. It does so in NettyServer constructor, however, you will still need to have this method hard coded. Possible that we just supply a list of handles to this initializer without fixing the parameters?